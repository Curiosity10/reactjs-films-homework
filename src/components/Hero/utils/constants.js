export const MAX_OVERVIEW_LENGTH = 220;
export const DEFAULT_MOVIE_RUNTIME = '0h 0m';
