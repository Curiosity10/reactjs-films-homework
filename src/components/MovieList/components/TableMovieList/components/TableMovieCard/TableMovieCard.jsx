import React from 'react';
import PropTypes from 'prop-types';
import styles from './TableMovieCard.scss';
import NumRating from '../../../../../NumRating';
import Genres from '../../../../../Genres';
import Button from '../../../../../Button';
import StarRating from '../../../../../StarRating';

const TableMovieCard = ({ movie }) => {
  const {
    rating,
    title,
    genres,
    overview,
    image,
  } = movie;
  const checkDescriptionLength = description => (
    description.length > 500 ? `${description.slice(0, 500)}...` : description
  );
  return (
    <article className={styles.card}>
      <img src={image} alt="Poster" />
      <div className={styles.info}>
        <h2 className={styles.title}>{title}</h2>
        <Genres genres={genres} />
        <p className={styles.overview}>{checkDescriptionLength(overview)}</p>
        <div className={styles.rating}>
          <StarRating starsSelected={rating} />
          <NumRating rating={rating} />
        </div>
        <Button type="primary">Watch Now</Button>
      </div>
    </article>
  );
};

TableMovieCard.propTypes = {
  movie: PropTypes.shape({
    rating: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    genres: PropTypes.array.isRequired,
    overview: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
};

export default TableMovieCard;
