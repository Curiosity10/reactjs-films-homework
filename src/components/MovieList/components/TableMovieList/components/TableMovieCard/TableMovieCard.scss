@import "../../../../../../styles/variables";

.card {
  width: 100%;
  box-sizing: border-box;
  padding: 30px;
  margin-bottom: 35px;
  background-color: #fff;
  position: relative;
  display: flex;
  box-shadow: 0 10px 40px 0 rgba(0, 0, 0, 0.3);

  & img {
    height: 320px;
    object-fit: cover;
    margin-right: 30px;
  }

  &_info {
    text-align: left;
  }

  &_title {
    color: #242b2d;
    font-size: 1.4rem;
    text-align: center;
    padding-top: 10px;
    margin-bottom: 20px;

    & ~ span {
      font-size: 1rem;
    }
  }

  &_overview {
    margin: 15px 0;
  }

  &_rating {
    margin-bottom: 15px;

    & span {
      margin-left: 15px;
      font-size: 0.9rem;
    }
  }
}

@media all and (max-width: $tablet-screen-size) {
  .card {
    flex-wrap: wrap;

    &_image {
      margin: 0 auto;

      & img {
        height: 300px;
        width: 100%;
      }
    }

    &_info {
      text-align: center;
    }

    &_title {
      font-size: 1.2rem;
      margin-bottom: 10px;

      & ~ span {
        font-size: 0.9rem;
      }
    }

    &_overview {
      font-size: 0.9rem;
    }

    &_rating {
      display: flex;
      justify-content: center;

      & span {
        font-size: 0.8rem;
      }
    }
  }
}
