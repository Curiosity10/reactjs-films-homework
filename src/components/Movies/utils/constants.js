export const MAX_OVERLAY_CARD_DESC_LENGTH = 180;
export const MAX_TABLE_CARD_DESC_LENGTH = 500;
export const MAX_SCROLL_HEIGHT = 0.99;
