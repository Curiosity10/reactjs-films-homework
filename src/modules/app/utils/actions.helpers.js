export const createRequestAction = requestName => ({
  type: requestName,
});

export const createFailureAction = (requestName, error) => ({
  type: requestName,
  payload: {
    error: error.message,
  },
});

export const createReceiveMoviesAction = ({
  type, data, page, genre = null, query = '',
}) => ({
  type,
  payload: {
    movies: data.results,
    currentPage: page,
    pagesCount: data.total_pages,
    genreId: genre,
    searchQuery: query,
  },
});

export const createAsyncAction = ({
  type, asyncFn, page = 1, genre = null, query = '', filterType = '',
}) => (dispatch) => {
  dispatch(createRequestAction(`${type}_REQUEST`));
  return asyncFn({ page, genre, query }, filterType)
    .then(response => response.json())
    .then(data => dispatch(createReceiveMoviesAction({
      type: `${type}_SUCCESS`, data, page, genre, query,
    })))
    .catch(error => dispatch(createFailureAction(`${type}_FAILURE`, error)));
};
