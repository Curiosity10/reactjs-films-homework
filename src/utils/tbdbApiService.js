import Movie from '../assets/json/movie.json';
import Genres from '../assets/json/genres.json';
import Popular from '../assets/json/popular.json';

const urlBase = 'https://image.tmdb.org/t/p/original';
const urlMinImg = 'https://image.tmdb.org/t/p/w300';

function getNormalRuntime(time) {
  const hours = Math.floor(time / 60);
  const minutes = time - 60 * hours;
  return `${hours}h ${minutes}m`;
}

function getMovieGenres(genresIds) {
  const genresArr = [];
  genresIds.map(genre => Genres.genres.map((item) => {
    if (item.id === genre) {
      return genresArr.push({ id: item.id, name: item.name });
    }
    return 0;
  }));
  return genresArr;
}

export function getMovieInfo() {
  return {
    title: Movie.original_title,
    genres: Movie.genres,
    rating: Movie.vote_average,
    runtime: getNormalRuntime(Movie.runtime),
    overview: Movie.overview,
    image: `${urlBase}${Movie.backdrop_path}`,
  };
}

export function getGenres() {
  return Genres.genres;
}

export function getPopularMovies() {
  return Popular.results.map(movie => ({
    id: movie.id,
    rating: movie.vote_average,
    title: movie.title,
    genres: getMovieGenres(movie.genre_ids),
    overview: movie.overview,
    image: `${urlMinImg}${movie.poster_path}`,
  }));
}
