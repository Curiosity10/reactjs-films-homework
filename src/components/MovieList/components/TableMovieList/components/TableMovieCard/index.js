import TableMovieCard from './TableMovieCard';

export default TableMovieCard;
