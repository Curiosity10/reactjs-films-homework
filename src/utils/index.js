import {
  getTopMovies,
  searchMovies,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo,
} from './api';
import { getMovies } from './tbdbApiService';
import truncate from './truncate';
import {
  createFailureAction,
  createRequestAction,
  createAsyncAction,
  createReceiveMoviesAction,
} from './actions.helpers';

export {
  getTopMovies,
  searchMovies,
  getGenres,
  getMoviesByGenre,
  getLatestMovies,
  getUpcomingMovies,
  getMovieVideo,
  truncate,
  getMovies,
  createFailureAction,
  createRequestAction,
  createAsyncAction,
  createReceiveMoviesAction,
};
