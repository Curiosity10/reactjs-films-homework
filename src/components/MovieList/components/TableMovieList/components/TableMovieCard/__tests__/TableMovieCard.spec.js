import React from 'react';
import { create } from 'react-test-renderer';
import TableMovieCard from '../index';
import { getMovieInfo } from '../../../../../../../utils/tbdbApiService';

describe('TableMovieCard component renders correctly', () => {
  const movie = { ...getMovieInfo() };

  it('TableMovieCard renders correctly', () => {
    const tree = create(
      <TableMovieCard
        movie={movie}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
