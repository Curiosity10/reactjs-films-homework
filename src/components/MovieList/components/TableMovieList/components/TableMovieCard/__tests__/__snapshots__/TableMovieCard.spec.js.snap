// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableMovieCard component renders correctly TableMovieCard renders correctly 1`] = `
<article
  className="card"
>
  <img
    alt="Poster"
    src="https://image.tmdb.org/t/p/original/59qyrVGLLtrcuFtJGdixzn7H4OI.jpg"
  />
  <div
    className="info"
  >
    <h2
      className="title"
    >
      Ð¢-34
    </h2>
    <span
      className="genre"
    >
      War
    </span>
    <span
      className="genre"
    >
      Drama
    </span>
    <span
      className="genre"
    >
      Adventure
    </span>
    <span
      className="genre"
    >
      Action
    </span>
    <p
      className="overview"
    >
      1941 - WWii. the second lieutenant Nikolai ivushkin, commander of a t-34, engages in an unequal battle against the tank ace Klaus Jager in a battle near moscow. His mission is more of a suicide - to destroy a dozen german tanks, all by himself. that said, luck does favour the bold. He wins the battle, barely survives, but loses his tank and lands himself in captivity for three long years... there was little to no chance for ivushkin and Jager to meet again, but the war knows how to throw a curve...
    </p>
    <div
      className="rating"
    >
      <div
        className="star"
      />
      <div
        className="star"
      />
      <div
        className="star"
      />
      <div
        className="star"
      />
      <div
        className="star"
      />
      <span
        className="numRating"
      >
        4.7
      </span>
    </div>
    <button
      aria-label="Watch Now"
      className="btn primary"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
  </div>
</article>
`;
